tests/cecil-tests/Helper.cs

```diff
@@ -105,6 +147,15 @@ public static string GetBCLDirectory (string assembly)
 			}
 		}
 
+		public static IEnumerable Net6PlatformAssemblies {
+			get {
+				yield return new TestCaseData (Path.Combine (Configuration.GetDotNetRoot (), "Microsoft.iOS.Ref", "ref", "net6.0", "Xamarin.iOS.dll"));
+				yield return new TestCaseData (Path.Combine (Configuration.GetDotNetRoot (), "Microsoft.tvOS.Ref", "ref", "net6.0", "Xamarin.TVOS.dll"));
+				yield return new TestCaseData (Path.Combine (Configuration.GetDotNetRoot (), "Microsoft.macOS.Ref", "ref", "net6.0", "Xamarin.Mac.dll"));
+				yield return new TestCaseData (Path.Combine (Configuration.GetDotNetRoot (), "Microsoft.MacCatalyst.Ref", "ref", "net6.0", "Xamarin.MacCatalyst.dll"));
+			}
```

Comment:
Getting all the ref assemblies seems like a fairly common thing to want to do in our tests, so I think it makes sense to implement this functionality in the Configuration class.

Here's a very quick version (and completely untested!): https://gist.github.com/rolfbjarne/3e54d364f38b4143bc66b82b15e6bffa

And then you'd call `Configuration.GetRefLibraries ()` to get all these libraries.

This also makes it easier to keep the code up-to-date if we ever add more platforms.

Reply:
Should I rework this similar existing code in the same way as well? https://github.com/xamarin/xamarin-macios/blob/573798ec38e2199352b57f5ad9233aaaae8180d1/tests/cecil-tests/Helper.cs#L134

Reply:
No, that can stay, it's for legacy Xamarin and will probably be removed in a couple of years anyway.

Reply:
sounds good, thanks for clarifying!

Reply:
There are some places in the gist where you referenced `targetFramework` when you're not able to. 
I tried to determine when you might have meant to refer to `platform` or potentially `platform.ToFramework()` instead, but I wasn't able to quite figure it all out. Could you help clarify my confusion here?

Reply:
Ah, I guess you mean this block: https://github.com/xamarin/xamarin-macios/pull/13590#discussion_r771678394

that can be simplified to:

```csharp
public static string GetRefDirectory (ApplePlatform platform)
{
    return Path.Combine (GetDotNetRoot (), GetRefNuGetName (platform), "ref", "net6.0");
}
````

and for this line:

https://gist.github.com/rolfbjarne/3e54d364f38b4143bc66b82b15e6bffa#file-foo-diff-L42

it's this:

```csharp
yield return Path.Combine (GetRefDirectory (platform), GetBaseLibraryName (platform));
```